pkg/types/spec/validations.go

```diff
@@ -828,6 +919,19 @@ func FindDuplicateNames(apis []userconfig.API) []userconfig.API {
 	return nil
 }
 
+func checkDuplicateModelNames(modelResources []*userconfig.ModelResource) error {
+	var names []string
```

Comment:
I'd still go with a slice because the names have to be extracted from the struct and then you also have to keep a record of the duplicate model. Then you'd have to compare the lengths.